onnxscript/optimizer/remove_unused_ir.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+from __future__ import annotations
+
+import logging
+
+import onnx
+
+from onnxscript import ir
+
+logger = logging.getLogger(__name__)
+
+
+def remove_unused_optional_outputs(
+    node: ir.Node, graph_outputs: set[ir.Value], onnx_opset_version: int
+) -> None:
+    try:
+        if node.domain not in {"", "onnx.ai"}:
+            return
+        op_schema = onnx.defs.get_schema(node.op_type, onnx_opset_version, domain=node.domain)
+    except Exception:
+        return
+
+    if node.op_type == "BatchNormalization":
+        # BatchNormalization op has 3 outputs: Y, running_mean, running_var
+        # If running_mean and running_var are not used, remove them, and the training_mode attribute
+        def is_used_output(i: int) -> bool:
+            if i < len(node.outputs):
+                val = node.outputs[i]
+                return val in graph_outputs or val.uses()
+            return False
+
+        if is_used_output(1) or is_used_output(2):
+            return
+        node.outputs[1].name = ""
+        node.outputs[2].name = ""
```

Comment:
The IR does not seem to allow us to remove entries from node.outputs. (The alternative seems to be to create a new node to replace the existing node.) For now, hope this works.